packages/vc-status-list/src/impl/encoding/cbor.ts

```diff
@@ -0,0 +1,160 @@
+import { DecodedStatusListPayload, resolveIdentifier } from './common'
+import { BitsPerStatus } from '@sd-jwt/jwt-status-list/dist'
+
+const cbor = com.sphereon.cbor
+const kmp = com.sphereon.kmp
+const decompressRawStatusList = (StatusList as any).decodeStatusList.bind(StatusList)
+
+const CWT_CLAIMS = {
+  SUBJECT: 2,
+  ISSUER: 1,
+  ISSUED_AT: 6,
+  EXPIRATION: 4,
+  TIME_TO_LIVE: 65534,
+  STATUS_LIST: 65533,
+} as const
+
+export const createSignedCbor = async (
+  context: IRequiredContext,
+  statusList: StatusList,
+  issuerString: string,
+  id: string,
+  keyRef?: string,
+): Promise<SignedStatusListData> => {
+  const identifier = await resolveIdentifier(context, issuerString, keyRef)
+
+  const encodeStatusList = statusList.encodeStatusList()
+  const compressedList = deflate(encodeStatusList, { level: 9 })
+  const compressedBytes = new Int8Array(compressedList)
+
+  const statusListMap = new cbor.CborMap(
+    kotlin.collections.KtMutableMap.fromJsMap(
+      new Map<com.sphereon.cbor.CborString, com.sphereon.cbor.CborItem<any>>([
+        [new cbor.CborString('bits'), new cbor.CborUInt(kmp.LongKMP.fromNumber(statusList.getBitsPerStatus()))],
+        [new cbor.CborString('lst'), new cbor.CborByteString(compressedBytes)],
+      ]),
+    ),
+  )
+
+  const protectedHeader = new cbor.CborMap(
+    kotlin.collections.KtMutableMap.fromJsMap(
+      new Map([[new cbor.CborUInt(kmp.LongKMP.fromNumber(16)), new cbor.CborString('statuslist+cwt')]]), // "type"
+    ),
+  )
+  const protectedHeaderEncoded = cbor.Cbor.encode(protectedHeader)
+  const claimsMap = buildClaimsMap(id, issuerString, statusListMap)
+  const claimsEncoded = cbor.Cbor.encode(claimsMap)
+
+  const signedCWT = await context.agent.keyManagerSign({
+    keyRef: identifier.kmsKeyRef,
+    data: claimsEncoded,
+    encoding: undefined,
+  })
+
+  const protectedHeaderEncodedInt8 = new Int8Array(protectedHeaderEncoded)
+  const claimsEncodedInt8 = new Int8Array(claimsEncoded)
+  const signatureInt8 = new Int8Array(signedCWT.signature)
+
+  const cwtArray = new cbor.CborArray(
+    kotlin.collections.KtMutableList.fromJsArray([
+      new cbor.CborByteString(protectedHeaderEncodedInt8),
+      new cbor.CborByteString(claimsEncodedInt8),
+      new cbor.CborByteString(signatureInt8),
+    ]),
+  )
+
+  const cwtEncoded = cbor.Cbor.encode(cwtArray)
+  const cwtBuffer = Buffer.from(cwtEncoded)
+  return {
+    statusListCredential: base64url.encode(cwtBuffer),
+    encodedList: base64url.encode(compressedList as Buffer), // JS in @sd-jwt/jwt-status-list drops it in like this, so keep the same method
+  }
+}
+
+function buildClaimsMap(
+  id: string,
+  issuerString: string,
+  statusListMap: com.sphereon.cbor.CborMap<com.sphereon.cbor.CborString, com.sphereon.cbor.CborItem<any>>,
+) {
+  const exp = Math.floor(new Date().getTime() / 1000)
+  const ttl = 65535 // FIXME figure out what value should be / come from and what the difference is with exp
```

Comment:
Please figure out what the difference is. Otherwise create a ticket and link it here

Reply:
@nklomp 
>ttl: OPTIONAL.  The ttl (time to live) claim, if present, MUST
      specify the maximum amount of time, in seconds, that the Status
      List Token can be cached by a consumer before a fresh copy SHOULD
      be retrieved.  The value of the claim MUST be a positive number
      encoded in JSON as a number.
      
Do we want to use / implement this now?

Reply:
=> Answer in Slack was no